src/algorithms/pid/Tools.h

```diff
@@ -0,0 +1,246 @@
+//
+// Common functions for PID algorithms
+// Several methods ported from Juggler's JugPID `IRTAlgorithmServices`
+//
+
+#pragma once
+
+// general
+#include <map>
+#include <math.h>
+#include <spdlog/spdlog.h>
+
+// ROOT
+#include <TVector3.h>
+
+// DD4hep
+#include <Evaluator/DD4hepUnits.h>
+
+// data model
+#include <edm4eic/CherenkovParticleIDCollection.h>
+#include <edm4hep/ParticleIDCollection.h>
+
+
+namespace eicrecon {
+
+  // Tools class, filled with miscellanous helper functions
+  class Tools {
+    public:
+
+      // -------------------------------------------------------------------------------------
+
+      // h*c constant, for wavelength <=> energy conversion [GeV*nm]
+      static constexpr double HC = dd4hep::h_Planck * dd4hep::c_light / (dd4hep::GeV * dd4hep::nm);
+
+
+      // -------------------------------------------------------------------------------------
+      // Radiator IDs
+
+      static std::unordered_map<int,std::string> GetRadiatorIDs() {
+        return std::unordered_map<int,std::string>{
+          { 0, "Aerogel" },
+          { 1, "Gas" }
+        };
+      }
+
+      static std::string GetRadiatorName(int id) {
+        std::string name;
+        try { name = GetRadiatorIDs().at(id); }
+        catch(const std::out_of_range& e) {
+          throw std::runtime_error(fmt::format("RUNTIME ERROR: unknown radiator ID={} in algorithms/pid/Tools::GetRadiatorName",id));
+        }
+        return name;
+      }
+
+      static int GetRadiatorID(std::string name) {
+        for(auto& [id,name_tmp] : GetRadiatorIDs())
+          if(name==name_tmp) return id;
+        throw std::runtime_error(fmt::format("RUNTIME ERROR: unknown radiator '{}' in algorithms/pid/Tools::GetRadiatorID",name));
+        return -1;
+      }
+
+
+      // -------------------------------------------------------------------------------------
+      // PDG mass lookup
+
+      // local PDG mass database
+      // FIXME: cannot use `TDatabasePDG` since it is not thread safe; until we
+      // have a proper PDG database service, we hard-code the masses we need;
+      // use Tools::GetPDGMass for access
+      static std::unordered_map<int,double> GetPDGMasses() {
+        return std::unordered_map<int,double>{
+          { 11,   0.000510999 },
+          { 211,  0.13957     },
+          { 321,  0.493677    },
+          { 2212, 0.938272    }
+        };
+      }
+
+      static double GetPDGMass(int pdg) {
+        double mass;
+        try { mass = GetPDGMasses().at(std::abs(pdg)); }
+        catch(const std::out_of_range& e) {
+          throw std::runtime_error(fmt::format("RUNTIME ERROR: unknown PDG={} in algorithms/pid/Tools::GetPDGMass",pdg));
+        }
+        return mass;
+      }
+
+      static int GetNumPDGs() { return GetPDGMasses().size(); };
+
+
+      // -------------------------------------------------------------------------------------
+      static std::vector<std::pair<double, double>> ApplyFineBinning(
```

Comment:
Somehow, all trivial functions ended up having (redundant) docstrings, but these two unobvious functions don't have explanation for their purpose

Reply:
I documented the new ones I wrote, but forgot to document the ones directly copied from Juggler. Will fix.